core/src/test/scala/org/apache/spark/scheduler/EventLoggingListenerSuite.scala

```diff
@@ -251,6 +261,233 @@ class EventLoggingListenerSuite extends SparkFunSuite with LocalSparkContext wit
+      logData.close()
+    }
+  }
+
+  /** Create a stage submitted event for the specified stage Id. */
+  private def createStageSubmittedEvent(stageId: Int) = {
+    SparkListenerStageSubmitted(new StageInfo(stageId, 0, stageId.toString, 0,
+      Seq.empty, Seq.empty, "details"))
+  }
+
+  /** Create a stage completed event for the specified stage Id. */
+  private def createStageCompletedEvent(stageId: Int) = {
+    SparkListenerStageCompleted(new StageInfo(stageId, 0, stageId.toString, 0,
+      Seq.empty, Seq.empty, "details"))
+  }
+
+  /** Create an executor added event for the specified executor Id. */
+  private def createExecutorAddedEvent(executorId: Int) = {
+    SparkListenerExecutorAdded(0L, executorId.toString, new ExecutorInfo("host1", 1, Map.empty))
+  }
+
+  /** Create an executor added event for the specified executor Id. */
+  private def createExecutorRemovedEvent(executorId: Int) = {
+    SparkListenerExecutorRemoved(0L, executorId.toString, "test")
+  }
+
+  /** Create an executor metrics update event, with the specified executor metrics values. */
+  private def createExecutorMetricsUpdateEvent(
+      executorId: Int,
+      executorMetrics: ExecutorMetrics): SparkListenerExecutorMetricsUpdate = {
+    val taskMetrics = TaskMetrics.empty
+    taskMetrics.incDiskBytesSpilled(111)
+    taskMetrics.incMemoryBytesSpilled(222)
+    val accum = Array((333L, 1, 1, taskMetrics.accumulators().map(AccumulatorSuite.makeInfo)))
+    SparkListenerExecutorMetricsUpdate(executorId.toString, accum, Some(executorMetrics))
+  }
+
+  /** Check that the two ExecutorMetrics match */
+  private def checkExecutorMetrics(
+      executorMetrics1: Option[ExecutorMetrics],
+      executorMetrics2: Option[ExecutorMetrics]) = {
+    (executorMetrics1, executorMetrics2) match {
+      case (Some(e1), Some(e2)) =>
+        assert(e1.timestamp === e2.timestamp)
+        assert(e1.jvmUsedHeapMemory === e2.jvmUsedHeapMemory)
+        assert(e1.jvmUsedNonHeapMemory === e2.jvmUsedNonHeapMemory)
+        assert(e1.onHeapExecutionMemory === e2.onHeapExecutionMemory)
+        assert(e1.offHeapExecutionMemory === e2.offHeapExecutionMemory)
+        assert(e1.onHeapStorageMemory === e2.onHeapStorageMemory)
+        assert(e1.offHeapStorageMemory === e2.offHeapStorageMemory)
+        assert(e1.onHeapUnifiedMemory === e2.onHeapUnifiedMemory)
+        assert(e1.offHeapUnifiedMemory === e2.offHeapUnifiedMemory)
+        assert(e1.directMemory === e2.directMemory)
+        assert(e1.mappedMemory === e2.mappedMemory)
+      case (None, None) =>
+      case _ =>
+        assert(false)
+    }
+  }
+
+  /** Check that the Spark history log line matches the expected event. */
+  private def checkEvent(line: String, event: SparkListenerEvent): Unit = {
+    assert(line.contains(event.getClass.toString.split("\\.").last))
+    event match {
+      case executorMetrics: SparkListenerExecutorMetricsUpdate =>
+        JsonProtocol.sparkEventFromJson(parse(line)) match {
```

Comment:
you can pull `JsonProtocol.sparkEventFromJson(parse(line))` out to avoid repeating, along with the type comparison.

```scala
val parsed = JsonProtocol.sparkEventFromJson(parse(line))
assert(parsed.getClass === event.getClass)
event match {
 ...
```

(also `assertTypeError` does something else entirely: http://doc.scalatest.org/2.2.6/#org.scalatest.Assertions)

Reply:
Thanks, modified.